pkg/sql/show_tenant.go

```diff
@@ -93,100 +110,145 @@ func (n *showTenantNode) getTenantName(params runParams) (roachpb.TenantName, er
 }
 
 func (n *showTenantNode) startExec(params runParams) error {
-	tenantName, err := n.getTenantName(params)
+	if n.all {
+		names, err := GetAllTenantNames(params.ctx, params.p.execCfg, params.p.Txn())
+		if err != nil {
+			return err
+		}
+		n.names = names
+	}
+
+	return nil
+}
+
+func getReplicationStats(
+	params runParams, job *jobs.Job,
+) (*streampb.StreamIngestionStats, *hlc.Timestamp, error) {
+	mgr, err := params.p.EvalContext().StreamManagerFactory.GetStreamIngestManager(params.ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	details, ok := job.Details().(jobspb.StreamIngestionDetails)
+	if !ok {
+		return nil, nil, errors.Newf("job with id %d is not a stream ingestion job", job.ID())
+	}
+	stats, err := mgr.GetStreamIngestionStats(params.ctx, details, job.Progress())
+	var protectedTimestamp hlc.Timestamp
 	if err != nil {
-		return err
+		// An error means we don't have stats but we can still present some info,
+		// therefore we don't fail here.
+		// TODO(lidor): we need a better signal from GetStreamIngestionStats(), instead of
+		// ignoring all errors.
+		log.Infof(params.ctx, "stream ingestion stats unavailable for tenant %q and job %d",
+			details.DestinationTenantName, job.ID())
+	} else {
+		if stats.IngestionDetails.ProtectedTimestampRecordID == nil {
+			// We don't have the protected timestamp record, but we still want to show
+			// the info we do have about tenant replication status, logging an error
+			// and continuing.
+			log.Warningf(params.ctx, "protected timestamp unavailable for tenant %q and job %d",
+				details.DestinationTenantName, job.ID())
+		} else {
+			ptp := params.p.execCfg.ProtectedTimestampProvider
+			record, err := ptp.GetRecord(params.ctx, params.p.Txn(), *stats.IngestionDetails.ProtectedTimestampRecordID)
+			if err != nil {
+				// Protected timestamp might not be set yet, no need to fail.
+				log.Warningf(params.ctx, "protected timestamp unavailable for tenant %q and job %d: %v",
+					details.DestinationTenantName, job.ID(), err)
+				return stats, nil, nil
+			}
+			protectedTimestamp = record.Timestamp
+		}
 	}
+	return stats, &protectedTimestamp, nil
+}
+
+func (n *showTenantNode) getTenantValues(
+	params runParams, tenantName roachpb.TenantName,
+) (*tenantValues, error) {
 	tenantRecord, err := GetTenantRecordByName(params.ctx, params.p.execCfg, params.p.Txn(), tenantName)
```

Comment:
As a heads up, there is another PR in flight to make most of these commands work with either IDs _or_ names. Not sure there is anything we want to do here other than being willing to help resolve the conflict when it occurs :D.